### PR TITLE
feat(stepfunctions): Improve aws stepfunctions support

### PR DESCRIPTION
**Pull request by idonava**
Add support to the following functions: send_task_success, send_task_heartbeat, describe_execution
Add support to step function lambda triggered by SQS

### CODE REVIEW

File: epsagon/events/botocore.py [hunk 1]
```diff
@@ -1380,6 +1401,34 @@ def process_start_exec_response(self, response):
...
+            ''
+        )
+
+    def process_describe_execution_response(self, response):
+        """
+        Process describeExecution response
+        :param response: response from Step function Client
+        :return: None
+        """
+        self.resource['metadata']['Execution ARN'] = response.get(
```
Comment (henprtz) [inline epsagon/events/botocore.py]: what happens if this is None?

File: epsagon/events/botocore.py [hunk 2]
```diff
@@ -1380,6 +1401,34 @@ def process_start_exec_response(self, response):
...
+            'executionArn')
+        self.resource['metadata']['Input'] = response.get('input')
+        self.resource['metadata']['Name'] = response.get('name')
+        self.resource['metadata']['Output'] = response.get('output')
+        self.resource['metadata']['Start Date'] = response.get('startDate')
+        self.resource['metadata']['Status'] = response.get('status')
+        stop_date = response.get('stopDate')
+        if stop_date:
+            self.resource['metadata']['Stop Date'] = stop_date
+
```
Comment (henprtz) [inline epsagon/events/botocore.py]: why are you only checking if stop date exists?

File: epsagon/utils.py [hunk 3]
```diff
@@ -286,3 +290,45 @@ def get_traceback_data_from_exception(exception):
...
+    :return: Title Case string
+    """
+    if not isinstance(camel_case_string, str):
+        return None
+    title_case = re.sub('([^-])([A-Z][a-z-]+)', r'\1 \2', camel_case_string)\
+        .title()
+    return title_case
+
+
+def add_data_to_resource_metadata(resource, dictionary, key):
```
Comment (ronyis) [inline epsagon/utils.py]: *`add_metadata_from_dict` ?

File: epsagon/utils.py [hunk 4]
```diff
@@ -286,3 +290,45 @@ def get_traceback_data_from_exception(exception):
...
+
+def parse_json(json_string):
+    """
+    Parse JSON string to a Python Dictionary
+    :param json_string: JSON string
+    :return: Python Dictionary
+    """
+    try:
+        return json.loads(json_string)
+    except ValueError:
```
Comment (ronyis) [inline epsagon/utils.py]: That's not the exception type when JSON load fails (check it)

**Pull request merged**